ui/app/components/app/transaction-list/transaction-list.component.js

```diff
@@ -5,20 +5,24 @@ import {
   nonceSortedCompletedTransactionsSelector,
   nonceSortedPendingTransactionsSelector,
 } from '../../../selectors/transactions';
+import { getCurrentChainId } from '../../../selectors';
 import { useI18nContext } from '../../../hooks/useI18nContext';
 import TransactionListItem from '../transaction-list-item';
 import Button from '../../ui/button';
 import { TOKEN_CATEGORY_HASH } from '../../../helpers/constants/transactions';
-import { SWAPS_CONTRACT_ADDRESS } from '../../../../../shared/constants/swaps';
+import { SWAPS_CHAINID_CONTRACT_ADDRESS_MAP } from '../../../../../shared/constants/swaps';
 import { TRANSACTION_TYPES } from '../../../../../shared/constants/transaction';
 
 const PAGE_INCREMENT = 10;
 
-const getTransactionGroupRecipientAddressFilter = (recipientAddress) => {
+const getTransactionGroupRecipientAddressFilter = (
+  recipientAddress,
+  chainId,
+) => {
   return ({ initialTransaction: { txParams } }) => {
     return (
       txParams?.to === recipientAddress ||
-      (txParams?.to === SWAPS_CONTRACT_ADDRESS &&
+      (txParams?.to === SWAPS_CHAINID_CONTRACT_ADDRESS_MAP[chainId] &&
```

Comment:
[Q] It's been a while since I viewed this code, and regret not having added documentation to it when I touched it last. If I understand correctly this is to help filter out transactions that aren't to the specified token. So when we're on, let's say DAI's asset page, we only see transactions interacting with DAI's contract address and swaps's contract address as long as the swap interacted with DAI's contract address?

I was able to walk through that logic in the current develop branch without the addition of chainId. If my assumption is correct it might be worth documenting the filter logic so that we don't lose the intent of these filters as logic changes.

Reply:
>  So when we're on, let's say DAI's asset page, we only see transactions interacting with DAI's contract address and swaps's contract address as long as the swap interacted with DAI's contract address?

yes, exactly